spec/authentication.mdwn

```diff
@@ -146,12 +146,46 @@ The response body will be a single JSON object with the following properties.
 
 URLs are returned only after logging in. This allows different URLs to be used for users located in different geographic datacentres within the same service.
 
 Note, if authentication is done via IP or mobile subscriber ID or some similar mechanism, a `201` response MAY be returned in response to the initial request (with just the username and client info).
 
+##### Downloading an attachment through a signed request
+
+Sometimes, authenticating a GET request using the usual `Authorization` header is not easily applicable. In such situations, a client SHOULD use a signed request to download the attachment. A signed request is an unauthenticated GET request containing a special query string parameter (a so-called `token`). The process below describes how a client can obtain such a token, then use it to download the file:
+
+* The client makes an authenticated POST request to the interpolated `downloadUrl` (see above for how to obtain this URL), that is, with all variables substituted with actual values. If authorization is granted, the server MUST send back a `text/plain` response with the signing `token` as the response body.
+* The client downloads the file by making an unauthenticated GET request to a URL formed by appending a parameter `token=<response to the previous request>` to the query part (see [RFC 3986](https://tools.ietf.org/html/rfc3986#section-3.4) for details) of the `downloadUrl`. If no query part was present in the `downloadUrl` before this operation, a new query part is created. The server MUST send back the contents of the requested attachment, as if it was requested through an authenticated GET request on the raw `downloadUrl`.
+
+The server SHOULD expire any given signing `token` quickly for obvious security reasons, but the actual expiration policy is up to the server implementation.
+  
+Sample HTTP exchange demonstrating the use of a signed request to download an attachment:
+
+* Obtain a token
+
+      POST /jmap/download/1234 HTTP/1.1
+      Host: server.jmap.io
+      Authorization: Bearer a3f5b6af745c9820264b6664d7c2c489adafd99f0f21a686c43aabfebf65418b
+
+      HTTP/1.1 200 OK
+      Content-Type: text/plain
+      Content-Length: 10
+
+      abcDEfGhIJ
```

Comment:
On github the headers are on the same line than the request, see: https://github.com/ddolcimascolo/jmap/blob/ee6319450f133f63e672047610a6bdc333112a80/spec/authentication.mdwn